File: Rooms.Core/Streams/RoomStreamOptions.cs
using System;
using System.IO;

namespace KolibSoft.Rooms.Core.Streams
{
    public sealed class RoomStreamOptions
    {

        public int MaxVerbLength { get; set; } = DefaultMaxVerbLength;
        public int MaxChannelLength { get; set; } = DefaultMaxChannelLength;
        public int MaxCountLength { get; set; } = DefaultMaxCountLength;
        public int MaxContentLength { get; set; } = DefaultMaxContentLength;
        public int MaxFastBuffering { get; set; } = DefaultMaxFastBuffering;
        public string TempContentFolderPath { get; set; } = DefaultTempContentFolderPath;

        public Stream CreateContentStream(long count)
        {
            if (count < 1) return Stream.Null;
            if (count <= MaxFastBuffering) return new MemoryStream((int)count);
            var path = Path.Combine(TempContentFolderPath, $"{DateTime.UtcNow.Ticks}");
            return new FileStream(path, FileMode.Create, FileAccess.ReadWrite, FileShare.None, 4096, FileOptions.DeleteOnClose);
        }

        public const int DefaultMaxVerbLength = 128;
        public const int DefaultMaxChannelLength = 32;
        public const int DefaultMaxCountLength = 32;
        public const int DefaultMaxContentLength = 4 * 1024 * 1024;
        public const int DefaultMaxFastBuffering = 1024 * 1024;
        public static readonly string DefaultTempContentFolderPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.InternetCache), "Content");

    }
}

File: Rooms.Web/wwwroot/src/streams/room_stream_options.js
class RoomStreamOptions {

    constructor(json = null) {
        this.readBuffering = json?.readBuffering ?? RoomStreamOptions.DEFAULT_READ_BUFFERING;
        this.writeBuffering = json?.readBuffering ?? RoomStreamOptions.DEFAULT_READ_BUFFERING;
        this.maxVerbLength = json?.maxVerbLength ?? RoomStreamOptions.DEFAULT_MAX_VERB_LENGTH;
        this.maxChannelLength = json?.maxChannelLength ?? RoomStreamOptions.DEFAULT_MAX_CHANNEL_LENGTH_LENGTH;
        this.maxCountLength = json?.maxCountLength ?? RoomStreamOptions.DEFAULT_MAX_COUNT_LENGTH;
        this.maxContentLength = json?.maxContentLength ?? RoomStreamOptions.DEFAULT_MAX_CONTENT_LENGTH_LENGTH;
    }

    static get DEFAULT_READ_BUFFERING() { return 1024; }
    static get DEFAULT_WRITE_BUFFERING() { return 1024; }
    static get DEFAULT_MAX_VERB_LENGTH() { return 128; }
    static get DEFAULT_MAX_CHANNEL_LENGTH() { return 32; }
    static get DEFAULT_MAX_COUNT_LENGTH() { return 32; }
    static get DEFAULT_MAX_CONTENT_LENGTH() { return 4 * 1024 * 1024; }

}

export {
    RoomStreamOptions
}